
syntax = "proto3";

message ShareSecret {
  string secret=1;
  bool active=2;
  //THis active will be True if the user has asked for password reset,
  //Once reset, the secret will be updated again as encryption with same key
  //is not good, New aes key will be generated with new salt
  string ownership = 3; //Public key of the user to whom this will be shared
  string secret_hash=4;
  string key=5;
  string role=6;
  uint32 idx=7;
  string reset_key=8;
  string created_on=9;
  string updated_on=10;
  bool executed=11; //Whether this has been executed in the present cycle after it was made active
  string executed_on=12; //The last time the acceptor has executed this contract i.e
                  //encrypted his allotted share with new reset key
  uint32 num_executions=13; //number of times the acceptor has executed this contract
  uint32 num_recoveries=14; //mber of times the user has receovered his/her mnemonic
  uint32 nonce=15;
  string nonce_hash=16;
  string signed_nonce=17;
  string public=18;

}
